sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -59,7 +60,8 @@ case class DataSourceAnalysis(conf: SQLConf) extends Rule[LogicalPlan] with Cast
       sourceAttributes: Seq[Attribute],
       providedPartitions: Map[String, Option[String]],
       targetAttributes: Seq[Attribute],
-      targetPartitionSchema: StructType): Seq[NamedExpression] = {
+      targetPartitionSchema: StructType,
+      conf: SQLConf): Seq[NamedExpression] = {
```

Comment:
Why do we need conf here, the constructor has conf though?

Reply:
Removed, thanks!